gcsio/src/main/java/com/google/cloud/hadoop/gcsio/testing/InMemoryGoogleCloudStorage.java

```diff
@@ -533,12 +534,10 @@ public GoogleCloudStorageItemInfo composeObjects(
         } while (bytesRead >= 0);
       }
     }
-
-    // If destination.hasGenerationId(), it'll automatically get enforced here by the create()
-    // implementation.
-    WritableByteChannel destChannel = create(destination, options);
-    destChannel.write(ByteBuffer.wrap(tempOutput.toByteArray()));
-    destChannel.close();
+    deleteObjects(ImmutableList.of(destination));
```

Comment:
if this fails, destination will be deleted, correct? can this take the system to an inconsistent state? Also, are we enforcing generationId?